samtranslator/open_api/open_api.py

```diff
@@ -101,7 +101,14 @@ def get_integration_function_logical_id(self, path_name, method_name):
 
         # Extract lambda integration (${LambdaName.Arn}) and split ".Arn" off from it
         regex = "([A-Za-z0-9]+\.Arn)"
-        match = re.findall(regex, arn)[0].split(".Arn")[0]
+        matches = re.findall(regex, arn)
+        # For Functions with AutoPublishAlias the integration URI will contain
+        # ${LambdaName}, not ${LambdaName.Arn} (both resolve to the AWS::Lambda::Alias ARN).
```

Comment:
It is not the LogicalId of the Lambda Resource but the LogicalId of the `AWS::Lambda::Alias` right? If you do `LambdaName.Arn` you get the ARN of LASTEST which is not the same ARN as from a `Ref: LambdaAliasResource`.

Reply:
Correct. The comment was misleading since `Alias` doesn't have `.Arn` like `Function` does (but `AliasLogicalId` and `FunctionLogicalId.Arn` both resolve to their respective ARNs).